src/Emanote/Model/Type.hs

```diff
@@ -78,6 +78,10 @@ modelReadyForView =
 inLiveServer :: Model -> Bool
 inLiveServer = Ema.CLI.isLiveServer . _modelEmaCLIAction
 
+modelRouteEncoderMust :: HasCallStack => Model -> RouteEncoder Model SiteRoute
+modelRouteEncoderMust model =
+  fromMaybe (error "RouteEncoder not initialized") $ model ^. modelRouteEncoder
```

Comment:
Also not a fan of this hack.

Reply:
Perhaps a way without hacks is to wrap the rendering functions in a `Reader (RouteEncoder Model SiteRoute)` monad? Then you can get the encoder at the start and don't bother with parameters. You would have to change more lines, though.

Reply:
Using `ModelT f` does the trick, at the expense of slight complexity.